report_py3o/static/src/js/py3oactionmanager.js

```diff
@@ -1,38 +1,75 @@
 /* Copyright 2017-2018 ACSONE SA/NV
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
-odoo.define('report_py3o.report', function (require) {
 
-var ActionManager = require('web.ActionManager');
+ odoo.define('report_py3o.report', function (require) {
+'use strict';
 
-ActionManager.include({
-    _executeReportAction: function (action, options) {
-        // Py3o reports
-        if ('report_type' in action && action.report_type === 'py3o' ) {
-            return this._triggerDownload(action, options, 'py3o');
-        } else {
-            return this._super.apply(this, arguments);
-        }
-    },
+    var ActionManager = require('web.ActionManager');
+    var core = require('web.core');
+    var crash_manager = require('web.crash_manager');
+    var framework = require('web.framework');
 
-    _makeReportUrls: function(action) {
-        var reportUrls = this._super.apply(this, arguments);
-        reportUrls.py3o = '/report/py3o/' + action.report_name;
+    
+    var trigger_download = function (session, response, c, action, options) {
+        return session.get_file({
+            url: '/report/download',
+            data: {data: JSON.stringify(response)},
+            complete: framework.unblockUI,
+            error: c.rpc_error.bind(c),
+            success: function () {
+                if (action && options && !action.dialog) {
+                    options.on_close();
+                }
+            },
+        });
+    };
+    
+    var make_report_url = function (action) {
+        var report_urls = {
+            'py3o': '/report/py3o/' + action.report_name,
+        };
         // We may have to build a query string with `action.data`. It's the place
         // were report's using a wizard to customize the output traditionally put
         // their options.
-        if (_.isUndefined(action.data) || _.isNull(action.data) ||
-            (_.isObject(action.data) && _.isEmpty(action.data))) {
+        if (_.isUndefined(action.data) || _.isNull(action.data) || (_.isObject(action.data) && _.isEmpty(action.data))) {
```

Comment:
@tbaden PLZ avoid to reformat the code.  Code reformatting adds burden to the review process without added value. Moreover, if the code looks better to you when formatted in this way, for me that reduce de readability...

Reply:
copied from v11 core, will refractor